crates/amaru-ledger/src/summary/governance.rs

```diff
@@ -76,111 +104,44 @@ impl DRepsSummary {
                             .count();
-                    dreps.insert(
-                        drep.clone(),
+
+                    (
+                        drep,
                         DRepState {
+                            registered_at,
                             metadata: anchor,
-                            mandate: last_interaction
-                                + DREP_EXPIRY
-                                + epochs_without_active_proposals as u64,
+                            mandate: Some(
+                                last_interaction
+                                    + DREP_EXPIRY
+                                    + epochs_without_active_proposals as u64,
+                            ),
+                            stake: 0, // NOTE: The actual stake is filled later when computing the
+                                      // stake distribution.
                         },
-                    );
-
-                    (drep, registered_at)
+                    )
                 },
             )
             .collect::<BTreeMap<_, _>>();
 
-        let delegations = db
-            .iter_accounts()?
-            .filter_map(|(credential, account)| {
-                account.drep.and_then(|(drep, since)| {
-                    if let Some(registered_at) = dreps_registrations.get(&drep) {
-                        if since >= *registered_at {
-                            // This is a registration with a previous registration of this DRep, it must be renewed
-                            return Some((credential, drep));
-                        }
-                    }
-                    None
-                })
-            })
-            .collect::<BTreeMap<StakeCredential, DRep>>();
-
-        Ok(DRepsSummary { delegations, dreps })
-    }
-}
-
-impl serde::Serialize for DRepsSummary {
-    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
-        let mut s = serializer.serialize_struct("DRepsSummary", 2)?;
-
-        let mut delegations = BTreeMap::new();
-        let mut keys = BTreeMap::new();
-        let mut scripts = BTreeMap::new();
-        self.delegations.iter().for_each(|(credential, drep)| {
-            if let Some(v) = into_drep_id(drep) {
-                match credential {
-                    StakeCredential::AddrKeyhash(hash) => keys.insert(*hash, v),
-                    StakeCredential::ScriptHash(hash) => scripts.insert(*hash, v),
-                };
-            }
-        });
-        delegations.insert("keys".to_string(), keys);
-        delegations.insert("scripts".to_string(), scripts);
-        s.serialize_field("delegations", &delegations)?;
-
-        let mut dreps = BTreeMap::new();
-        self.dreps.iter().for_each(|(drep, st)| {
-            if let Some(id) = into_drep_id(drep) {
-                dreps.insert(id, st);
-            }
-        });
-        s.serialize_field("dreps", &dreps)?;
+        let default_drep_state = || DRepState {
```

Comment:
Is this closure to avoid the clone?

Reply:
More like to avoid repeating the same code twice. Ideally, I'd just `DRepState::default()`. Not sure why it stayed as a closure here.